<launch>
  <!-- launch turtlebot3_autorace_extrinsic_camera_calibration.launch before launch this file -->

  <arg name="calibration_mode" default="$(env AUTO_DT_CALIB)" doc="calibration_mode type [calibration, action]"/>

  <!-- level crossing detection -->
  <node pkg="turtlebot3_autorace_detect" type="detect_level" name="detect_level" output="screen">
    <rosparam command="load" file="$(find turtlebot3_autorace_detect)/param/level/level.yaml" />
    <param if="$(eval calibration_mode == 'calibration')" name="/is_detection_calibration_mode" value="True"/>
    <param if="$(eval calibration_mode == 'action')" name="/is_detection_calibration_mode" value="False"/>    
    <remap from="/detect/image_input" to="/camera/image_compensated" />
    <remap from="/detect/image_input/compressed" to="/camera/image_compensated/compressed" />
    <remap from="/detect/image_output" to="/detect/image_level" />
    <remap from="/detect/image_output/compressed" to="/detect/image_level/compressed" />
    <remap from="/detect/image_output_sub1" to="/detect/image_level_color_filtered" />
    <remap from="/detect/image_output_sub1/compressed" to="/detect/image_level_color_filtered/compressed" />
  </node>
</launch>